### ISSUE TITLE
Does the learning rate change during training?

### ISSUE THREAD

**Issue by danieljf24** [fchollet/keras#1145]
rms = RMSprop()
model.compile(loss='categorical_crossentropy', optimizer=rms)

model.fit(X_train, Y_train,
          batch_size=batch_size, nb_epoch=nb_epoch,
          show_accuracy=True, verbose=2,
          validation_data=(X_test, Y_test))

if I used the above optimizer or any other optimizer, does the learning rate change during training?
Shall I change the learning rate manually?

**Comment by rpinsler** [comment]
Yes, the learning rate usually changes during training, depending on the optimizer you choose. As no one wants to manually change the learning rate, there are different mechanisms (so called learning rate schedules) to automatically do this. Momentum is another trick to change the learning rate.

Furthermore, some optimizers such as SGD maintain a global learning rate across all parameters, whereas other approaches such as Adagrad or RMSprop effectively use per-parameter learning rates. A nice summary is given [here](http://cs231n.github.io/neural-networks-3/#update).

Check the [keras documentation](http://keras.io/optimizers/) for the default parameter values.

**Comment by suixudongi8** [comment]
Use SGD wit mom.
Yes, it will change, absolutely!

**Comment by danieljf24** [comment]
@rpinsler Thanks very much.
Could I use the real-time learning rate when the model is trained?

**Comment by suixudongi8** [comment]
In the Callbacks module, LearningRateScheduler can do what you want.
